lib/matplotlib/_types.py

```diff
@@ -0,0 +1,179 @@
+      center lines.
+    - *bevel* is the "squared-off" style. It can be thought of as a rounded
+      corner where the "circular" part of the corner has been cut off.
+
+    .. note::
+
+        The *miter* option can be controlled further by specifying a "miter
+        limit", which specifies how long a miter tip can get before it is
+        automatically "bevel"ed off. Matplotlib does not currently expose a
+        ``miterlimit`` parameter to the user, and most backends simply use the
+        upstream default value. For example, the PDF backend assumes the
+        default value of 10 specified by the PDF standard, while the SVG
+        backend does not even specify the miter limit, resulting in a default
+        value of 4 per the SVG specification.
+
+        A more detailed description of the effect of a miter limit can be found
+        in the `Mozilla Developer Docs
+        <https://developer.mozilla.org/en-US/docs/Web/SVG/Attribute/stroke-miterlimit>`_
+    """
+
+    miter = auto()
+    round = auto()
+    bevel = auto()
+
+    def __init__(self, s):
+        s = _deprecate_case_insensitive_join_cap(s)
+        Enum.__init__(self)
+
+    @staticmethod
+    def demo():
+        import numpy as np
+        import matplotlib.pyplot as plt
+
+        def plot_angle(ax, x, y, angle, style):
+            phi = np.radians(angle)
+            xx = [x + .5, x, x + .5*np.cos(phi)]
+            yy = [y, y, y + .5*np.sin(phi)]
+            ax.plot(xx, yy, lw=12, color='tab:blue', solid_joinstyle=style)
+            ax.plot(xx, yy, lw=1, color='black')
+            ax.plot(xx[1], yy[1], 'o', color='tab:red', markersize=3)
+
+        fig, ax = plt.subplots(figsize=(8, 6))
+        ax.set_title('Join style')
+        for x, style in enumerate(['miter', 'round', 'bevel']):
+            ax.text(x, 5, style)
+            for y, angle in enumerate([20, 45, 60, 90, 120]):
+                plot_angle(ax, x, y, angle, style)
+                if x == 0:
+                    ax.text(-1.3, y, f'{angle} degrees')
+        ax.set_xlim(-1.5, 2.75)
+        ax.set_ylim(-.5, 5.5)
+        ax.set_axis_off()
+        fig.show()
+JoinStyle.input_description = "{" \
+        + ", ".join([js.name for js in JoinStyle]) \
+        + "}"
+
+
+class CapStyle(str, _AutoStringNameEnum):
+    r"""
+    Define how the two endpoints (caps) of an unclosed line are drawn.
+
+    How to draw the start and end points of lines that represent a closed curve
+    (i.e. that end in a `~.path.Path.CLOSEPOLY`) is controlled by the line's
+    `JoinStyle`. For all other lines, how the start and end points are drawn is
+    controlled by the *CapStyle*.
+
+    For a visual impression of each *CapStyle*, `view these docs online
+    <CapStyle>` or run `CapStyle.demo`:
+
+    .. plot::
+        :alt: Demo of possible CapStyle's
+
+        from matplotlib._types import CapStyle
+        CapStyle.demo()
+
+    Available options:
```

Comment:
```suggestion
    Available values:
```
to match `CapStyle(value)` signature.

Not quite sure with "available". Alternatives: "Supported values:", "Values:"